src/Qowaiv/Text/CharBuffer.cs

```diff
@@ -1,57 +1,112 @@
+internal static class CharBuffer
 {
-    internal static readonly int NotFound = -1;
+    /// <summary>Gets the max length of an array.</summary>
+    /// <remarks>
+    /// Array.MaxLength is not available for .NET standard 2.0.
+    /// </remarks>
+    private const int Array_MaxLength = 0X7FFFFFC7;
 
-    private readonly char[] buffer;
-    private int start;
-    private int end;
+    private const int NotFound = -1;
 
-    /// <summary>Initializes a new instance of the <see cref="CharBuffer" /> class.</summary>
-    private CharBuffer(int capacity) => buffer = new char[capacity];
+    [Pure]
+    public static int BufferSize(this ReadOnlySpan<char> span)
+    {
+        var size = span.Length * 2;
+        return (uint)size > Array_MaxLength ? Array_MaxLength : size;
+    }
 
-    /// <summary>Initializes a new instance of the <see cref="CharBuffer" /> class.</summary>
-    public CharBuffer(string str) : this(str.Length) => Add(str);
+    [CollectionMutation]
+    public static int Unify(this ReadOnlySpan<char> span, Span<char> buffer)
+    {
+        var length = 0;
 
-    /// <summary>Gets the length of the buffer.</summary>
-    public int Length => end - start;
+        foreach (var ch in span)
+        {
+            if (!IsMarkup(ch))
+            {
+                var relace = DiacriticSearch.IndexOf(ch);
 
-    /// <summary>Gets <see cref="char" /> on the specified index.</summary>
-    public char this[int index]
-    {
-        get => buffer[index + start];
-        private set => buffer[index + start] = value;
+                if (relace != NotFound)
+                {
+                    buffer[length++] = char.ToUpperInvariant(DiacriticReplace[relace]);
+                }
+                else if (DiacriticLookup.TryGetValue(ch, out var chs))
+                {
+                    buffer[length++] = char.ToUpperInvariant(chs[0]);
+                    buffer[length++] = char.ToUpperInvariant(chs[1]);
+                }
+                else
+                {
+                    buffer[length++] = char.ToUpperInvariant(ch);
+                }
+            }
+        }
+        return length;
     }
 
-    /// <summary>Returns true if the buffer is empty.</summary>
-    [Pure]
-    public bool IsEmpty() => Length == 0;
-
-    /// <summary>Returns true if the buffer is not empty.</summary>
-    [Pure]
-    public bool NotEmpty() => !IsEmpty();
+    [CollectionMutation]
+    public static int ToNonDiacritic(this ReadOnlySpan<char> span, Span<char> buffer, string ignore = "")
+    {
+        var length = 0;
 
-    /// <summary>Gets the first <see cref="char" /> of the buffer.</summary>
-    [Pure]
-    public char First() => buffer[start];
+        foreach (var ch in span)
+        {
+            if (ignore.Contains(ch))
+            {
+                buffer[length++] = ch;
+            }
+            else
+            {
+                var relace = DiacriticSearch.IndexOf(ch);
```

Comment:
Performance-wise it would probably be better to use a `FrozenSet` (or for targets that don't support frozen collections, use a regular old `HashSet`). I understand it's easier to just define this set as a `string`, but since a `string` is an `IEnumerable<char>` anyway, you could just do `.ToFrozenSet()` or `.ToHashSet()` on the string.

Reply:
Edit: or use a `FrozenDictionary` instead to find the replacement.

Reply:
But based on the index we get the replacement. And a set will not tell you its order?!

Reply:
Hence the suggestion to use a dictionary

Reply:
@CptWesley  I've applied the suggestion of using a dictionary. Converting that one to a frozen dictionary will be handled when #465 is solved.